### PR TITLE
Vignettes: Demography

### PR DESCRIPTION
**Pull request by tbreweric**
The demography vignette has been updated to remove the dependency on ggplot2 by recreating the figures using base R plotting functions, add more detailed textual explanations of the examples in the vignette, and improve the structure of the vignette.

Figures:
The figures have been recreated using base R plotting functions and their colours brought in line with the scheme used across the vignette updates.

Code Chunks:
The code chunks have been tidied and commented more extensively to ensure readers can follow the steps.

Text:
An introduction section has been added to explain the purpose of the vignette and outline its contents. The vignette has been broadly divided into two sections, static and time-varying custom death rates. Within these sections, additional subsections have been included to help highlight a logical order of steps when running malariasimulation. Each section has had the textual explanation expanded to be more informative to the reader.

### CODE REVIEW

**Review by pwinskill — APPROVED** [review]
V clear - thank you!

File: vignettes/Demography.Rmd [hunk 1]
```diff
@@ -15,24 +15,38 @@ knitr::opts_chunk$set(
...
-suppressPackageStartupMessages(library(ggplot2))
+# Load in the requisite package(s):
 library(malariasimulation)
 ```
 
-# Parameterisation
+# Introduction
 
-We are going to create a scenario where we track severe prevalence for 5 years.
+The dynamics of malaria transmission, and the efficacy of interventions designed to interrupt it, are highly context specific and an important consideration is the demography of the population(s) under investigation. To suit the research needs of the user, `malariasimulation` allows for the specification of custom human population demographics by allowing them to specify age-group specific death rates parameters. It also enables users to instruct the model to render outputs with variables of interest (e.g. human population, number of cases detectable by microscopy, number of severe cases, etc.) presented by user-defined age groups. In this vignette we'll use simple, illustrative cases to demonstrate how to customise the output rendered by the `run_simulation()` function, specify both fixed and time-varying custom death rates.
```
Comment (htopazian) [inline vignettes/Demography.Rmd:24]: last sentence "and how to specify..."

File: vignettes/Demography.Rmd [hunk 2]
```diff
@@ -41,116 +55,186 @@ simparams <- get_parameters(
...
 
-Let's run the simulations 
+## Simulations
+
+Having established parameter sets with both default and custom demographic parameterisations, we can now run a simulation for each using the `run_simulation()` function. We'll also add a column to each identify the runs.
 
 ```{r}
-# run and combine the outputs
+
+# Run the simulation with the default (?) demogaphic set-up
```
Comment (htopazian) [inline vignettes/Demography.Rmd:101]: take out "default (?)", this is now custom

File: vignettes/Demography.Rmd [hunk 3]
```diff
@@ -41,116 +55,186 @@ simparams <- get_parameters(
...
   .0234852, .0988317, .046755, .1638875, .1148753, .3409079, .2239224,
   .8338688) / 365
 
-demography_params <- set_demography(
-  demography_params,
+# Pass these custom ages and death rates through the in-built set_demography function 
+dem_params <- set_demography(
+  dem_params,
   agegroups = ages,
   timesteps = 0,
```
Comment (htopazian) [inline vignettes/Demography.Rmd:86]: could just add a little comment here to say that you are starting the custom demography at the beginning of the model run

File: vignettes/Demography.Rmd [hunk 4]
```diff
@@ -41,116 +55,186 @@ simparams <- get_parameters(
...
 
-deathrates <- c(
-  .4014834, .0583379, .0380348, .0395061, .0347255, .0240849, .0300902,
-  .0357914, .0443123, .0604932, .0466799, .0426199, .0268332, .049361,
-  .0234852, .0988317, .046755, .1638875, .1148753, .3409079, .2239224,
-  .8338688) / 365
-# Let's increase the death rates in some age groups
-deathrates_changed <- deathrates
-deathrates_changed[3:6] <- deathrates_changed[3:6] * 10
+We'll start by making a copy of the base parameters which have been instructed to output the population size through time (`simparams`). Next, we create a version of the `deathrates` vector created earlier updated to increase the death rates of the 5-10, 10-15, and 15-20 age groups by 1000%. Finally, we use `set_demography()` to update our parameter list and specify our custom demography. This is achieved providing the `timesteps` argument with a vector of days to update the death rates (0 and 730 days), and providing `deathrates` with a matrix of death rates where each row corresponds to a set of death rates for a given update.
```
Comment (htopazian) [inline vignettes/Demography.Rmd:179]: wait is this increasing the death rate by 10% (rather than 1000%)?

**Review by htopazian — APPROVED** [review]
Hey Tom, this is really very nice! I like the examples you have chosen and the plots. Just a few grammatical things, good to merge :)

File: vignettes/Demography.Rmd (hunk 1 above)
Comment (tbreweric) [reply vignettes/Demography.Rmd:24]: Corrected the the last sentence to: 

"In this vignette we'll use simple, illustrative cases to demonstrate how to customise the output rendered by the `run_simulation()` function, and how to specify both fixed and time-varying custom death rates."

File: vignettes/Demography.Rmd (hunk 2 above)
Comment (tbreweric) [reply vignettes/Demography.Rmd:101]: Corrected the comment to "# Run the simulation with the default demographic set-up" for the default run and left "# Run the simulation for the custom demographic set-up:" for the line which runs the simulation with the customised demography.

File: vignettes/Demography.Rmd (hunk 3 above)
Comment (tbreweric) [reply vignettes/Demography.Rmd:86]: I have updated the comment to:

"# Update the parameter list with the custom ages and death rates through the in-built
# set_demography() function, instructing the custom demography to be implemented at the
# beginning of the model run (timestep = 0):"

and included the following in the textual explanation:

"Here, we instruct the model to implement the changes to the demographic parameters at the beginning of the simulation (`timesteps = 0`)."

**Pull request merged**